src/serde_util.rs

```diff
@@ -0,0 +1,76 @@
+use core::fmt;
+use core::marker::PhantomData;
+use core::str::{self, FromStr};
+use serde::de;
+
+pub struct HexVisitor<T> {
```

Comment:
I'm probably the one at fault, but `HexVisitor` appears to be a horrible name for something that essentially relies on `FromStr` and thus only incidentally decodes hex.

Reply:
Good catch!

I guess it is really just a `FromStrVisitor` then?